src/common/modules/UnicodeFontHandler.js

```diff
@@ -0,0 +1,245 @@
+"use strict";
+
+import * as AddonSettings from "/common/modules/AddonSettings/AddonSettings.js";
+// import * as BrowserCommunication from "/common/modules/BrowserCommunication/BrowserCommunication.js";
+import { isMobile } from "./MobileHelper.js";
+
+import { COMMUNICATION_MESSAGE_TYPE } from "/common/modules/data/BrowserCommunicationTypes.js";
+import { fonts } from "/common/modules/data/Fonts.js";
+
+const caseIds = Object.freeze(["Lowercase", "Uppercase", "Capitalize Each Word", "Toggle Case"]);
+const fontIds = Object.freeze(["Superscript", "Small Caps", "All Small Caps", "Unicase", "separator", "Serif bold", "Serif italic", "Serif bold italic", "Sans-serif", "Sans-serif bold", "Sans-serif italic", "Sans-serif bold italic", "Script", "Script bold", "Fraktur", "Fraktur bold", "Monospace", "Double-struck", "separator", "Circled", "Circled (black)", "Squared", "Squared (black)", "Fullwidth"]);
+
+// Thunderbird
+const THUNDERBIRD = typeof messenger !== "undefined";
+
+// Chrome
+const CHROME = Object.getPrototypeOf(browser) !== Object.prototype;
+
+/**
+ * Change Unicode font.
+ *
+ * @param {string} atext
+ * @param {string} afont
+ * @returns {string}
+ */
+function changeFont(atext, afont) {
+	const font = fonts[afont];
+	// console.log(afont, font);
+	let output = '';
+
+	for (let letter of atext) {
+		const code = letter.charCodeAt(0);
+		if (code >= 33 && code <= 127) {
+			if (font.length == 94)
+				letter = font[code - '!'.charCodeAt(0)];
+			else if (letter >= 'A' && letter <= 'Z') {
+				letter = font[code - 'A'.charCodeAt(0)];
+			} else if (letter >= 'a' && letter <= 'z') {
+				if (font.length == 26 || font.length == 26 + 10)
+					letter = font[code - 'a'.charCodeAt(0)];
+				else if (font.length == 26 + 26 || font.length == 26 + 26 + 10)
+					letter = font[code - 'a'.charCodeAt(0) + 26];
+			} else if (letter >= '0' && letter <= '9') {
+				if (font.length == 26 + 10)
+					letter = font[code - '0'.charCodeAt(0) + 26];
+				else if (font.length == 26 + 26 + 10)
+					letter = font[code - '0'.charCodeAt(0) + 26 + 26];
+			}
+		}
+		output += letter;
+	}
+
+	return output;
+}
+
+/**
+ * Capitalize Each Word.
+ *
+ * @param {string} atext
+ * @returns {string}
+ */
+function capitalizeEachWord(atext) {
+	// Requires Firefox/Thunderbird 78
```

Comment:
…because/why?

(again loink Bugzilla issues if applicable)

Reply:
This uses regular expression Unicode property escapes and lookbehind assertions, which [requires Firefox/Thunderbird 78](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#Browser_compatibility). I will update the comment.